runtime/sema/type.go

```diff
@@ -3522,6 +3528,29 @@ func (t *CompositeType) GetCompositeKind() common.CompositeKind {
 	return t.Kind
 }
 
+func (t *CompositeType) getBaseCompositeKind() common.CompositeKind {
+	if t.Kind != common.CompositeKindAttachment {
+		return common.CompositeKindUnknown
+	}
+	switch base := t.baseType.(type) {
+	case *CompositeType:
+		return base.Kind
+	case *InterfaceType:
+		return base.CompositeKind
+	case *SimpleType:
+		if base == AnyResourceType {
+			return common.CompositeKindResource
+		} else if base == AnyStructType {
+			return common.CompositeKindStructure
+		}
```

Comment:
Maybe generalize to any simple type with 

```suggestion
		return base.CompositeKind()
```

And add a function 

```go
func (t *SimpleType) CompositeKind() common.CompositeKind {
	if t.IsResource {
		return common.CompositeKindResource
	} else {
		return common.CompositeKindStructure
	}
}
```

This will also allow us to easily extend simple types to be used for e.g. event declarations by refactoring this function, instead of having to look for all uses of all simple types and such "remote decision making"